File: task3/Elixir/Subtask3.exs
defmodule DiamondPattern do
  def print_diamond(n) when n > 0 do
    # Generate the upper part of the diamond
    for i <- 0..(n-1) do
      IO.puts(String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1))
    end

    # Generate the lower part of the diamond
    for i <- (n-2)..0 do
      IO.puts(String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1))
    end
  end

  def print_diamond(_n) do
    IO.puts("Please enter a positive integer.")
  end
end

# Prompt the user for input
IO.write("Enter the number of rows for the diamond: ")
n = IO.gets("") |> String.trim() |> String.to_integer()

# Print the diamond pattern
DiamondPattern.print_diamond(n)

File: task3/Ruby/Subtask3.rb
def print_diamond(n)
  diamond = ""

  # Upper part of the diamond
  (0...n).each do |i|
    diamond += " " * (n - i - 1) + "*" * (2 * i + 1) + "\n"
  end

  # Lower part of the diamond
  (n - 2).downto(0) do |i|
    diamond += " " * (n - i - 1) + "*" * (2 * i + 1) + "\n"
  end

  diamond
end

# Get the number from the user
print "Enter a number: "
n = gets.to_i

# Generate and print the diamond pattern
puts print_diamond(n)
